app/renderer/actions/Session.js

```diff
@@ -350,54 +350,77 @@ export function newSession (caps, attachSessId = null) {
         break;
     }
 
-    let rejectUnauthorized = !session.server.advanced.allowUnauthorized;
-    let proxy;
-    if (session.server.advanced.useProxy && session.server.advanced.proxy) {
-      proxy = session.server.advanced.proxy;
+    // TODO W2D handle proxy and rejectUnauthorized cases
+    //let rejectUnauthorized = !session.server.advanced.allowUnauthorized;
+    //let proxy;
+    //if (session.server.advanced.useProxy && session.server.advanced.proxy) {
+    //  proxy = session.server.advanced.proxy;
+    //}
+
+    dispatch({type: SESSION_LOADING});
+
+
+    const hostname = username && accessKey ? `${username}:${accessKey}@${host}` : host;
+    const serverOpts = {
+      hostname,
+      port: parseInt(port, 10),
+      protocol: https ? 'https' : 'http',
+      path,
+      connectionRetryCount: CONN_RETRIES,
+    };
+
+    // If a newCommandTimeout wasn't provided, set it to 0 so that sessions don't close on users
+    if (!desiredCapabilities.newCommandTimeout) {
+      desiredCapabilities.newCommandTimeout = 0;
+    }
+
+    // If someone didn't specify connectHardwareKeyboard, set it to true by
+    // default
+    if (typeof desiredCapabilities.connectHardwareKeyboard === 'undefined') {
+      desiredCapabilities.connectHardwareKeyboard = true;
     }
 
-    // Start the session
-    ipcRenderer.send('appium-create-new-session', {
+    let driver = null;
+    try {
+      if (attachSessId) {
+        driver = await Web2Driver.attachToSession(attachSessId, serverOpts);
+        driver._isAttachedSession = true;
+      } else {
+        driver = await Web2Driver.remote(serverOpts, desiredCapabilities);
+      }
+    } catch (err) {
+      showError(err, 0);
+      return;
+    } finally {
+      dispatch({type: SESSION_LOADING_DONE});
+      // Save the current server settings
+      await settings.set(SESSION_SERVER_PARAMS, session.server);
+    }
+
+    // The homepage arg in ChromeDriver is not working with Appium. iOS can have a default url, but
+    // we want to keep the process equal to prevent complexity so we launch a default url here to make
+    // sure we don't start with an empty page which will not show proper HTML in the inspector
+    const {browserName = ''} = desiredCapabilities;
+
+    if (browserName.toLowerCase() !== '') {
```

Comment:
toLowerCase is redundant here. Perhaps, `trim` would make more sense

Reply:
yeah this code was here before, but i'll fix it.